lib/src/code_generator/pointer.dart

```diff
@@ -9,7 +9,15 @@ import 'writer.dart';
 /// Represents a pointer.
 class PointerType extends Type {
   final Type child;
-  PointerType(this.child);
+
+  PointerType._(this.child);
+
+  factory PointerType(Type child) {
```

Comment:
Should we maybe have something more general to recognize "internal-dart-types" and return the "user-visible-dart-types"?

For example, if we have something more general we could also recognize `Pointer<Char>` and make it available as `String` in signatures?

It should probably be a separate pass. Rather than making a choice point in factory constructors of types.

```dart
  // Parse the bindings according to config object provided.
  final library = parse(config);

  final libraryTransformed = transform(library);

  // Generate file for the parsed bindings.
  final gen = File(config.output);
  libraryTransformed.generateFile(gen);
```

Or maybe because we already make choices in `parse` based on the config, it should a `Type wrapDartType(Type type)` function being invoked on any type. And for now it only replaces Pointer<CObject> with ObjCObjectPointer.